CONTRIBUTING.md

```diff
@@ -0,0 +1,99 @@
+# Contributing to KoLmafia
+
+First off, thank you for contributing your time and knowledge. This project
+would not exist if not for volunteers like you!
+
+## Some useful prerequisites (lifted from [Github docs](https://github.com/github/docs/blob/main/CONTRIBUTING.md))
+
+- [Set up Git](https://docs.github.com/en/get-started/quickstart/set-up-git)
+- [GitHub flow](https://docs.github.com/en/get-started/quickstart/github-flow)
+- [Collaborating with pull requests](https://docs.github.com/en/github/collaborating-with-pull-requests)
+
+All work must be done in feature branches (typically within your fork of the
+project), and reviewed by at least one of the core developers through Github's
+Pull Request feature.
+
+We strongly encourage that you provide tests as part of your pull request, as
+this helps us understand the problem at hand and avoid breaking your change in
+the future.
+
+## Guidelines (adapted from [our previous guide](https://kolmafia.us/threads/25364))
+
+(Editor's note: you can mentally replace all instances of "patch" with
+"pull request".)
+
+A successful submit requires social and technical skills.
+
+The set of social skills includes the ability to get along with devs at the
+KoLmafia.us community. Users that have managed to alienate themselves from
+members of the dev team tend to have their proposed patches ignored.
+
+The ability to determine the appropriateness of a patch and the level of support
+for it is extremely useful. A patch that violates the spirit of KoLmafia (for
+example, that allows access to non-KoLmafia directories in the user's file
+system) will be rejected. A patch that changes an existing feature will need
+some discussion. A patch that changes a lot of code to implement a feature of at
+best niche interest is unlikely to be accepted.
+
+The technical skills are the ability to write and test Java code that works in
+the context of KoLmafia. A prerequisite for success is the ability to read and
+understand KoLmafia's code and infer its architecture since those are not well
+documented in the traditional sense.
+
+Individuals with a good track record of creating and submitting patches can ask
+to be made a dev. The active devs who can do this are MCroft and gausie.
```

Comment:
Is this still true and what we want or how we expect to do things?  Can non-staff contribute to a review even if they cannot approve the PR?  I'd like the answer to be yes and part of the audition to become staff would include constructive participation in reviews.

Reply:
> Can non-staff contribute to a review even if they cannot approve the PR? I'd like the answer to be yes and part of the audition to become staff would include constructive participation in reviews.

Yes, agreed on both fronts. I also added a vague comment about how we may initiate contact, since IIRC that's how I became a dev.

What I don't want is to encourage a system of basically mini-mods where non-devs go around nitpicking others' PRs. For now "constructively engaging" seems good enough, but we can revisit if it becomes an issue.